File: Problemas/1352A - Sum of Round Numbers/1352A_Sum_Of_Round_Numbers.py
# 1352A - Sum of Round Numbers
# https://codeforces.com/problemset/problem/1352/A

# Input:
# 1) pruebas = Número de pruebas
pruebas = int(input())

# Realiza cada prueba
for prueba_actual in range(0,pruebas):

    # Input:
    # 2) numero = Número a descomponer
    numero = int(input())

    # Dígito actual (derecha a izquierda)
    digitos = 1

    # Lista de números redondos
    numeros_redondos_lista = []

    # Revisa que el número sea distinto a 0
    while numero != 0:

        # Saca el número módulo 10 ^ dígitos para extraer el número redondo (menor a mayor)
        numero_redondo_actual = numero % (10 ** digitos)

        # Si el número redondo actual es distinto de 0, entonces se tiene que incluir en la lista y restar del número 
        if numero_redondo_actual != 0:

            # Añade el número redondo a la lista de números redondos
            numeros_redondos_lista.append(numero_redondo_actual)

            # Resta el número redondo del número original
            numero = numero - numero_redondo_actual

        # Avanza al siguiente dígito
        digitos += 1

    # Imprime el número de números redondos encontrados
    print(len(numeros_redondos_lista))

    # Imprime los números redondos
    for numero_redondo in numeros_redondos_lista:
        print(numero_redondo, end=" ")
    
    # Imprime un espacio de línea para la siguiente prueba
    print()

File: Problemas/1352A - Sum of Round Numbers/1352A_Sum_Of_Round_Numbers.cpp
// 1352A - Sum of Round Numbers
// https://codeforces.com/problemset/problem/1352/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) prueba = Número de pruebas
    int pruebas;
    cin >> pruebas;

    // Realiza cada prueba
    for(int prueba_actual = 0; prueba_actual < pruebas; prueba_actual++) {

        // Input:
        // 2) numero = Número a descomponer
        int numero;
        cin >> numero;
        
        // Número de números redondos
        int numeros_redondos = 0;

        // Copia del número
        int numero_digitos = numero;

        // Revisa que el número sea distinto a 0
        while(numero_digitos != 0) {

            // Si el dígito es distinto de 0, entonces sumamos +1 números redondos
            if(numero_digitos % 10 != 0) {
                numeros_redondos++;
            }

            // Divide entre 10 para pasar al siguiente dígito
            numero_digitos = numero_digitos / 10;
        }

        // Posición decimal del número redondo (unidad, decena, centena, etc...)
        int decimal = 1;

        // Revisa que el número sea distinto a 0
        while(numero != 0) {
            
            // Si el dígito es distinto de 0, entonces extraemos el número redondo
            if(numero % 10 != 0) {
                
                // Extrae el número redondo
                int num_redondo = (numero % 10) * decimal;
                
                // Imprime el número redondo
                cout << num_redondo << " ";
            }
            
            // Divide entre 10 para pasar al siguiente dígito
            numero = numero / 10;

            // Multiplica por 10 para pasar a la siguiente unidad decimal
            decimal = decimal * 10;
        }
        cout << endl;
    }

    return 0;
}
